src/libs/ascent/runtimes/ascent_vtkh_data_adapter.cpp

```diff
@@ -127,6 +127,76 @@ void CopyArray(vtkm::cont::ArrayHandle<T> &vtkm_handle, const T* vals_ptr, const
   vtkm_handle = vtkm::cont::make_ArrayHandle(vals_ptr, size, copy);
 }
 
+
+template<typename T>
+void
+BlueprintIndexArrayToVTKmIdArray(const conduit::Node &n,
+                                 bool zero_copy,
+                                 vtkm::cont::ArrayHandle<T> &vtkm_handle)
+{
+    int array_size = n.dtype().number_of_elements();
+
+    if( sizeof(T) == 1 ) // uint8 is what vtk-m will use for this case.
+    {
+        if(n.is_compact() && n.dtype().is_uint8())
+        {
+            // directly compatible
+            const void *idx_ptr = n.data_ptr();
+            CopyArray(vtkm_handle, (const T*)idx_ptr, array_size,zero_copy);
+        }
+        else
+        {
+            // we need to convert to uint8 to match vtkm::Id
+            vtkm_handle.Allocate(array_size);
+            void *ptr = (void*) vtkh::GetVTKMPointer(vtkm_handle);
+            Node n_tmp;
+            n_tmp.set_external(DataType::uint8(array_size),ptr);
+            n.to_uint8_array(n_tmp);
+        }
+    }
+    else if( sizeof(T) == 2)
+    {
+        // unsupported! -- TODO ERROR!
```

Comment:
just flagging this spot as a reminder to put an error message

Reply:
resolved